src/character.cpp

```diff
@@ -2138,81 +2149,184 @@ hp_part Character::body_window( const std::string &menu_header,
-                                           get_effect_int( effect_disinfected, e.bp ) );
-        } else {
-            treated = false;
         }
+        int new_d_power = static_cast<int>( std::floor( disinfectant_power ) );
+        // this prevents false prediction of effects that overreach maximum limits
+        if( disinfected ) {
+            const effect &eff = get_effect( effect_disinfected, e.bp );
+            if( new_d_power > eff.get_max_intensity() ) {
+                new_d_power = eff.get_max_intensity();
+            }
 
+        }
         const int new_hp = clamp( current_hp + bonus, 0, maximal_hp );
-
-        std::stringstream msg;
-
-        const nc_color old_hp_col = has_any_effect ? all_state_col :
-                                    limb_is_broken ? c_dark_gray : c_green;
         const auto &aligned_name = std::string( max_bp_name_len - utf8_width( e.name ), ' ' ) + e.name;
-        msg << string_format( "<color_%s>%s</color> <color_%s>%s</color>",
+        msg << string_format( "<color_%s>%s</color> %s",
                               color_name( all_state_col ), aligned_name,
-                              color_name( old_hp_col ), hp_str( current_hp, maximal_hp ) );
+                              hp_str( current_hp, maximal_hp ) );
+        desc << string_format( "<color_%s>%s</color> %s",
+                               color_name( all_state_col ), e.name,
+                               hp_str( current_hp, maximal_hp ) );
         if( current_hp != new_hp || has_curable_effect ) {
-            const nc_color new_hp_col = has_any_effect ? new_state_col :
-                                        limb_is_broken ? c_dark_gray : c_green;
-            msg << string_format( " <color_dark_gray>-></color> <color_%s>%s</color>",
-                                  color_name( new_hp_col ), hp_str( new_hp, maximal_hp ) );
+            msg << string_format( " <color_yellow>-></color> %s", hp_str( new_hp, maximal_hp ) );
+            desc << string_format( " <color_yellow>-></color> %s", hp_str( new_hp, maximal_hp ) ) << "\n";
+        } else {
+            desc << "\n";
         }
-        if( treated ) {
-            msg << string_format( " <color_%s>%s</color>",
-                                  color_name( all_state_col ), treatment_str );
+
+        if( limb_is_broken ) {
+            desc << tag_colored_string( _( "It is broken. It needs a splint or surgical attention." ),
+                                        c_red ) << "\n";
         }
 
-        bmenu.addentry( i, true, MENU_AUTOASSIGN, msg.str() );
-    }
+        // BLEEDING block
+        if( bleeding ) {
+            const int bleed_int = get_effect_int( effect_bleed, e.bp );
+            desc << "<color_red>" << string_format( "%s: ", get_effect( effect_bleed, e.bp ).get_speed_name() );
+            switch( bleed_int ) {
+                case 1:
+                    desc << _( "It is bleeding." );
```

Comment:
`Bleeding: It is bleeding` sounds slightly odd. Maybe shorten this string (and others below too) to something like `Bleeding: Weak` or `Bleeding: Mild`?